anchorecli/cli/utils.py

```diff
@@ -642,47 +642,37 @@ def format_output(config, op, params, payload):
-                service_status = "N/A"
-                if service_record['status']:
-                    service_status = "up"
+                if service_record.get('status', None):
+                    service_status = 'up'
                 else:
-                    service_status = "down ({})".format(service_record['status_message'])
+                    service_status = 'down ({})'.format(service_record['status_message'])
 
-                outlist.append("Service "+service_record['servicename']+" ("+service_record['hostid']+", " +service_record['base_url'] +"): " + str(service_status))
-                if not db_version:
-                    try:
-                        db_version = service_record['service_detail']['db_version']
-                    except:
-                        pass
+                out_list.append('Service {} ({}, {}): {}'.format(service_record['servicename'],
+                                                                 service_record['hostid'],
+                                                                 service_record['base_url'],
+                                                                 str(service_status)))
 
+                # This is a fallback mechanism to get the db & code versions from a non-api service
+                # (should there be no healthy api service available)
+                if not db_version:
+                    db_version = service_record.get('service_detail', {}).get('db_version', None)
                 if not code_version:
-                    try:
-                        code_version = service_record['service_detail']['version']
-                    except:
-                        pass
-
-                # override with any discovered API service that is up
-                try:
-                    if service_record['servicename'] == 'apiext' and service_status == 'up':
-                        api_code_version = service_record['service_detail']['version']
-                        code_version = api_code_version
-                except:
-                    pass
+                    code_version = service_record.get('service_detail', {}).get('version', None)
 
-            for k in outlist:
-                obuf = obuf + k + "\n"
-            obuf = obuf + "\n"
+                # Set the code & db versions with the details from the first discovered API service that is up
+                if service_record.get('servicename', '') == 'apiext' and service_status == 'up':
+                    service_detail = service_record.get('service_detail', {})
+                    code_version = service_detail.get('version', None)
+                    db_version = service_detail.get('db_version', None)
 
-            obuf = obuf + "Engine DB Version: {}\n".format(db_version)
-            obuf = obuf + "Engine Code Version: {}\n".format(code_version)
+            output_buffer = '\n'.join(out_list)
+            output_buffer += '\n\nEngine DB Version: {}\n'.format(db_version)
```

Comment:
`db_version`  has potential to be a `None` here. Would this be OK to read as:

```
Engine DB Version: None
```

Or would it be useful to have something like `"Not Found"` ?